google-cloud-storage/src/main/java/com/google/cloud/storage/ParallelCompositeUploadBlobWriteSessionConfig.java

```diff
@@ -523,6 +520,41 @@ public static PartNamingStrategy prefix(String prefixPattern) {
       return new WithPrefix(rand, prefixPattern);
     }
 
+    @BetaApi
+    public static PartNamingStrategy objectLevelPrefix() {
+      return objectLevelPrefix("");
+    }
+
+    /**
+     * Strategy in which an explicit stable prefix with object name included is present on each part
+     * and intermediary compose object.
+     *
+     * <p>General format is
+     *
+     * <pre><code>
+     *   {prefixPattern}/{objectName}/{randomKeyDigest};{objectInfoDigest};{partIndex}.part
+     * </code></pre>
+     *
+     * <p>{@code {objectInfoDigest}} will be fixed for an individual {@link BlobWriteSession}.
+     *
+     * <p><b><i>NOTE:</i></b>The way in which both {@code randomKeyDigest} and {@code
+     * objectInfoDigest} are generated is undefined and subject to change at any time.
+     *
+     * <p>Care must be taken when choosing to specify a stable prefix as this can create hotspots in
+     * the keyspace for object names. See <a
+     * href="https://cloud.google.com/storage/docs/request-rate#naming-convention">Object Naming
```

Comment:
I don't think this objectLevelPrefix is needed as a prefix to object name. Reading through https://cloud.google.com/storage/docs/request-rate#randomness_after_a_common_prefix_is_effective_under_the_prefix; the default empty string impl would be enough along with the randomization that it uses.